src/main/java/controllers/GameController.java

```diff
@@ -0,0 +1,137 @@
+package controllers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import controllers.data_objects.CardDisplayData;
+import controllers.data_objects.GameDisplayData;
+import controllers.interfaces.GameBridge;
+import enums.Rank;
+import enums.Suit;
+import enums.TurnAction;
+import ui.interfaces.GameUI;
+import use_cases.CardResponseModel;
+import use_cases.PlayerGameInputBoundary;
+import use_cases.PlayerGameRequestModel;
+import use_cases.PlayerGameResponseModel;
+
+/**
+ * The controller class for when a User is playing a Game.
+ */
+public class GameController implements GameBridge {
+    private final PlayerGameInputBoundary inputBoundary;
+
+    private GameUI UI;
+    private Suit selectedSuit;
+    private Rank selectedRank;
+    private String currentPlayer;
+
+    /**
+     * Construct a Controller for the interaction between a User/Player and a Game.
+     * Only one should be created for any particular Game instance.
+     * @param inputBoundary The inputBoundary which takes in a request and returns a response.
+     */
+    public GameController(PlayerGameInputBoundary inputBoundary) {
+        this.inputBoundary = inputBoundary;
+    }
+
+    /**
+     * Requests the state of the initial game.
+     */
+    @Override
+    public void requestStartData() {
+        PlayerGameResponseModel response = getResponse(null, null, null, TurnAction.START);
+        GameDisplayData data = getGameDisplayData(response);
+        updateCurrentPlayer(response);
+        UI.updateView(data);
+    }
+
+    /**
+     * Requests to play the selected card.
+     */
+    @Override
+    public void requestPlay() {
+        PlayerGameResponseModel response = getResponse(currentPlayer, selectedSuit, selectedRank, TurnAction.PLAY);
+        GameDisplayData data = getGameDisplayData(response);
+        updateCurrentPlayer(response);
+        UI.updateView(data);
```

Comment:
Hmm, I don't know about having outer layer (UI) code inside the controllers. Since this is an interface, I'd put this in the layer/package with the controllers.

Reply:
fair point